src/gui/widgets/FadeButton.cpp

```diff
@@ -82,21 +105,35 @@ void FadeButton::customEvent( QEvent * )
 void FadeButton::paintEvent( QPaintEvent * _pe )
 {
 	QColor col = m_normalColor;
+
 	if( ! m_stateTimer.isNull() && m_stateTimer.elapsed() < FadeDuration )
 	{
+		// The first part of the fade, when a note is triggered.
 		const float state = 1 - m_stateTimer.elapsed() / FadeDuration;
-		const int r = (int)( m_normalColor.red() *
+		const int r = (int)( m_holdColor.red() *
 					( 1.0f - state ) +
 			m_activatedColor.red() * state );
-		const int g = (int)( m_normalColor.green() *
+		const int g = (int)( m_holdColor.green() *
 					( 1.0f - state ) +
 			m_activatedColor.green() * state );
-		const int b = (int)( m_normalColor.blue() *
+		const int b = (int)( m_holdColor.blue() *
 					( 1.0f - state ) +
 			m_activatedColor.blue() * state );
 		col.setRgb( r, g, b );
 		QTimer::singleShot( 20, this, SLOT( update() ) );
 	}
+	else if( ! m_stateTimer.isNull()
+		&& m_stateTimer.elapsed() >= FadeDuration
+		&& activeNotes > 0)
+	{
+		// The fade is done, but at least one note is still held.
+		col = m_holdColor;
+	}
+	else
+	{
+		// No fade, no notes. Reset to default color.
+		col = m_normalColor;
```

Comment:
It seems like there's no secondary fade out corresponding to R in ADSR. Are there any reasons not implementing this?

Reply:
Not really, mostly just because I didn't quite understand the fade code and didn't think it was too necessary.

It has been implemented now.